File: plat_du_jour.js
const webdriver = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');
const {Builder, By, Key, until} = require('selenium-webdriver');
const options = new chrome.Options();
options.addArguments('--headless');
options.addArguments('--disable-dev-shm-usage');
options.addArguments('--no-sandbox');
options.addArguments('--disable-setuid-sandbox');

async function getMenu() {
    let menu = [];
    let element = await driver.findElement(webdriver.By.xpath("//ul[@class='meal_foodies']"));
    let text = await element.getText();
    menu = text.split('\n');
    return menu;
}

let driver = new webdriver.Builder()
    .forBrowser('chrome')
    .setChromeOptions(options)
    .build();

let main_url = "https://www.crous-montpellier.fr/restaurant/resto-u-triolet-2/";

driver.get(main_url);

// driver.findElement(webdriver.By.xpath("//time[@class='menu_date_title']")).getText().then((text) => {
//    let date_du_menu = text.split();
//    // rest of the code
// });

// let date_du_jour = (new Date().toISOString().slice(0,10).split("-"))[2];

if(true){
    getMenu().then(menu => {

        let liste_menu = ["Entrée chaude", "Entrées froides", "Cuisine du monde", "Pastaria", "Saveurs de la mer 1", "Plat des régions", "Rôtisserie", "Saveurs de la mer 2", "Desserts"];
    
        let tous_menus = {};
        let menu_current = [];
    
        for (let plat of menu) {
            console.log(plat)
            if (liste_menu.includes(plat)) {
                if (menu_current.length>0) {
                    let keys = Object.keys(tous_menus);
                    tous_menus[keys[keys.length-1]] = menu_current;
                    menu_current=[];
                }
                tous_menus[plat]=[];
            } else {
                if (plat === "(plat végétarien)" || plat === "(plat unique)") {
                    menu_current[menu_current.length-1] = menu_current[menu_current.length-1]+" "+plat;
                } else {
                    menu_current.push(plat);
                }
            }
        }
        let keys = Object.keys(tous_menus);
        tous_menus[keys[keys.length-1]] = menu_current;
        menu_current=[];

    console.log(tous_menus);
    });
}

File: plat_du_jour.py
import datetime
import json
import time

from notify_run import Notify
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

def contain_or(sentence):
    list_word=sentence.split()
    if "Ou" in list_word:
        if len(list_word)==1:
            return 1
        else:
            return 2
    else:
        return 0

def get_current_menu():

    options = webdriver.ChromeOptions()
    options.add_argument('--headless')
    driver = webdriver.Chrome('chromedriver.exe',options=options)
    driver.maximize_window()

    tous_menus = {}
    menu_current = []
    try:
        main_url = "https://www.crous-montpellier.fr/restaurant/cafet-space-2/"
        time.sleep(1)
        driver.get(main_url)

        date_du_jour = datetime.datetime.now()
        jour_semaine = date_du_jour.weekday()
        print("jul")
        time.sleep(1)
        menu = driver.find_element_by_xpath("//ul[@class='meal_foodies']")
        titre_plat = menu.find_elements_by_xpath("li")
        menu = (menu.text).split("\n")
        liste_menu = []
        for i in range(len(titre_plat)):
            liste_menu.append((titre_plat[i].text).split('\n')[0])
        for plat in menu:
            if plat in liste_menu:
                if len(menu_current)>0:
                    keys = list(tous_menus.keys())
                    tous_menus[keys[-1]] = menu_current
                    menu_current=[]
                tous_menus.update({plat:[]})
            else:
                if plat == "(plat végétarien)" or plat == "(plat unique)" or plat == "(plat unique végétarien )" or contain_or(plat)==2:
                    menu_current[-1] = menu_current[-1]+" "+plat
                else:
                    menu_current.append(plat)
        tous_menus[liste_menu[-1]]=menu_current
    except Exception as e:
        print(e)
        tous_menus = -1

    print(tous_menus)
    return tous_menus